spec/chef/knife/ec_error_handler_spec.rb

```diff
@@ -0,0 +1,30 @@
+require File.expand_path(File.join(File.dirname(__FILE__), "..", "..", "spec_helper"))
+require 'chef/knife/ec_error_handler'
+require 'fakefs/spec_helpers'
+
+def net_exception(code)
+  s = double("status", :code => code.to_s)
+  Net::HTTPServerException.new("I'm not real!", s)
+end
+
+describe Chef::Knife::EcErrorHandler do
+  let(:dest_dir) { Dir.mktmpdir }
+
+  before(:each) do
+    @knife_ec_error_handler = described_class.new(dest_dir, Class)
+  end
+
+  it "#display" do
+    @knife_ec_error_handler.add(net_exception(123))
+    expect { @knife_ec_error_handler.display }.to output(/Error Summary Report/).to_stdout
+    expect { @knife_ec_error_handler.display }.to output(/I'm not real!/).to_stdout
+    expect { @knife_ec_error_handler.display }.to output(/Net::HTTPServerException/).to_stdout
+  end
+
+  it "#add" do
```

Comment:
I think @marcparadise is getting at some of the behavior that should be validated, here. The test might want to verify that the `msg` is being written to the correct file, for example. I'm also wondering if it's worth verifying that the correct `req_path` and `req_method` come through in a situation where they're available. As is, this test still seems to be testing `display`

I always have to re-lookup how to handle time in specs, and it's probably not the most important thing to verify here, but I believe you can stub `Time.now`, maybe.